### PR TITLE
New responsive classes

### PR DESCRIPTION
**Pull request by pablohpsilva**
Currently there is no way to hide an element on a breakpoint and above, eg.: medium size and above. Given this issue, a good enhancement would be having some responsive classes. I created and I'm proposing the following classes:

```scss
md-hide-#{type}-and-up
```

Where **#{type}** can be: xsmall, small, medium, large, xlarge.

The media queries that controls those classes are:

```scss
@mixin layout-xsmall-and-up {
  @media (min-width: #{$breakpoint-xsmall - 300px}) { // 300px
    @content;
  }
}

@mixin layout-small-and-up {
  @media (min-width: #{$breakpoint-small - 300px}) { // 660px
    @content;
  }
}

@mixin layout-medium-and-up {
  @media (min-width: #{$breakpoint-small - 16px}) { //  944px
    @content;
  }
}

@mixin layout-large-and-up {
  @media (min-width: #{$breakpoint-medium - 16px}) { // 1264px
    @content;
  }
}

@mixin layout-xlarge-and-up {
  @media (min-width: #{$breakpoint-large - 16px}) { // 1904px
    @content;
  }
}
```

### Why?

Like I said before, if I want to show an element only on medium screens, right now, there is no way of doing it, based on the documentation: 

> Hide a layout container/child on screen sizes **less than** or equal to given breakpoint. 

I had to implement it today to show/hide a fab button. I think it will be a good asset to the project.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 .DS_Store
 node_modules/
 npm-debug.log
+dist
```
Comment (marcosmoura) [inline .gitignore]: The dist folder needs to be commited in the release script.

**Comment by marcosmoura** [comment]
Hi @pablohpsilva 

I liked it a lot. It will be a great feature. I just want to discuss one thing: 
This should be applied to flex and other options too. Like: md-flex-medium-and-up or md-column-small-and-up.

What do you think? Can you do this?

**Comment by pablohpsilva** [comment]
Hello @marcosmoura 

Thank you! I think I can do this. It's kinda late right now, but I will work on that ASAP.

**Comment by pablohpsilva** [comment]
Good morning @marcosmoura . Now that I'm thinking straight, I could not see a reason to add those classes you asked for other than just having it hanging around.

My thought process was:

> If I have a class that will hide an element no matter what happens given the screen size, why should I bother on having auxiliaries classes that would do exactly the same with a different name?

What do you think about it?

**Comment by marcosmoura** [comment]
No. Sorry. I think I wasn't that clear. My idea is:
If we have new classes to hide elements to breakpoints greater than a given breakpoint, why not having classes to apply flex sizes and to change columns to rows? I mean, if I want to develop a mobile first application, I would use those new classes/parameters. Did you get it?

**Comment by pablohpsilva** [comment]
Oohhh... That's brilliant @marcosmoura !
Alright, I can do this. Could you please accept this PR? I'll create another one with the other classes.

**Comment by marcosmoura** [comment]
Yep! Approved and Merged!

**Comment by pablohpsilva** [comment]
@marcosmoura Will you merge the develop branch on master soon?
I'm just asking (so I can remove my helpers css and use vue-material's)...

**Comment by marcosmoura** [comment]
@pablohpsilva No. The master changed a lot due to some build errors. I will merge the develop only after the 0.7.0 release.

**Comment by primadi** [comment]
@pablohpsilva there is exist md-hide-#{type}, why not create md-show-#{type} ? right now I am using my own css for my simple usage..

```
.small-visible {
  @media (min-width: 500px) {
    display: none;
  }
}

.small-hide {
  @media (max-width: 500px) {
    display: none;
  }
}
```

**Comment by pablohpsilva** [comment]
Hello @primadi ! How are you?
If you want to hide something using vue-material, you can use `md-hide-medium`, for example. That class will hide all elements on xs, sm, and medium screen. This PR was made to create the opposite effect. Classes like `md-hide-medium-and-up` will hide all elements on md, lg screens.

I got what you're saying. You can create an issue asking this enhancement or make a PR yourself! :D 

Feel free to choose. Thank you for your help, @primadi

**Comment by primadi** [comment]
Hello @pablohpsilva 

I have 2 md-icon, one has class `md-hide-medium`, this icon hide on **1280px and down**.
the other icon has class `md-hide-medium-and-up`, this icon hide on **960px and up**.

Is this intended behaviour ?

Thanks.

**Comment by pablohpsilva** [comment]
Yes, @primadi you're right :D 
for `md-hide-medium` you have the rule (**max-width**: 1280px)
for the `md-hide-medium-and-up` you have the rule (**min-width**: 960px)

**Pull request merged**